chapters/modules/chapter_8/pages/sharp.adoc

```diff
@@ -0,0 +1,67 @@
+
+== What is the SHARP System?
+
+The SHARP system, which stands for "Shared Prover," is a mechanism used in StarkNet that aggregates multiple Cairo programs from different users, each containing different logic. These Cairo programs are then executed together, generating a single proof that is common to all the programs. This proof is then sent to Ethereum, where it is verified by a Solidity verifier on-chain.
+
+The main advantage of the SHARP system is its ability to reduce costs and improve efficiency in the StarkNet network. By aggregating multiple Cairo jobs, the protocol can take advantage of the exponential acceleration provided by the STARK proof system. As a result, each individual transaction costs much less, making the entire process more affordable for users.
+
+Additionally, the SHARP system allows smaller users with limited computation to benefit from joining other jobs and sharing the cost of generating the proofs. This collaborative approach is similar to using public transportation instead of a private car, where the cost is distributed among all participants, making it more affordable for everyone.
+
+== Recursive Proofs in the SHARP System
+
+One of the most interesting aspects of the SHARP system is its use of recursive proofs. Instead of sending the generated proofs directly to the Solidity verifier, the proofs are first sent to a Cairo verifier, which is a STARK verifier written in Cairo. The Cairo verifier then checks the proofs off-chain and generates a new proof, which can be sent back to the SHARP system.
+
+This process continues in a recursive manner, with each new proof being sent to the Cairo verifier until a trigger is reached. At this point, the last proof in the series is sent to the Solidity verifier on Ethereum. This approach allows for greater parallelization of the computation and reduces the time and cost associated with generating and verifying proofs.
+
+At first glance, recursive proofs may seem more complex and time-consuming. However, there are several benefits to this approach:
+
+. *Parallelization*: Recursive proofs allow for parallelization of work, reducing latency for users and improving system efficiency.
+. *Lower cloud costs*: Since each job is shorter, the required memory for processing is reduced, resulting in lower cloud costs.
+. *Optimization*: Recursive proofs enable the system to optimize for various factors, including latency, on-chain costs, and time to proof.
+
+== SHARP Backend Architecture and Data Pipeline
+
+The SHARP backend architecture consists of several services that work together to process Cairo jobs and generate proofs. These services include the gateway, validator, scheduler, Cairo runner, prover, and dispatcher.
+
+. *Gateway*: Cairo jobs enter the system through the gateway.
```

Comment:
Like an API service?

Reply:
Great question; Kineret did not explain this part. I will ask here in slack.